core/EE_Session.core.php

```diff
@@ -885,7 +890,10 @@ private function _save_session_to_db($clear_session = false)
     {
         // don't save sessions for crawlers
         // and unless we're deleting the session data, don't save anything if there isn't a cart
-        if ($this->request->isBot() || (! $clear_session && ! $this->cart() instanceof EE_Cart)) {
+        if (apply_filters(
+            'FHEE__EE_Session___save_session_to_db__abort_session_save',
+            $this->request->isBot() || (! $clear_session && ! $this->cart() instanceof EE_Cart)
```

Comment:
I would prefer the filter to be added as an additional condition after the `EE_Cart` instance check so that we only check the filter if everything else evaluates to `true`, as any other result means the conditional will immediately pass and enter the return block..

The way it is in this commit means it will **always** check the filter which is unnecessary if any of the other conditions are true, ie:

 - if the current request is being made by a bot... we don't need to check the filter
 - if we are attempting to clear the session... we don't need to check the filter
 - if the session cart property contains a valid cart... we don't need to check the filter

we **only** need to check the filter if none of those conditions apply but we still want to save the session.

Reply:
I prefer it this way as it's more flexible (other plugins can completely override the default logic in both to force either true or false, whereas your suggestion only allows them to make it false). And to my understanding, running a filter isn't onerous (we run thousands of them). iirc this is what I've been doing everywhere I want to allow code to override an `if` statement, so this seems consistent... yes, even if our current understanding dictates that, theoretically, we shouldn't need to override the `if` statement to force it to be true.

Is efficiency the main reason you're making this recommendation? Or is there something else I'm missing?

Reply:
The way I worded it sounds like my concern is efficiency but no, it's actually because we don't want those other conditions being overridden.

I guess the bot issue is debatable... but I don't know why someone would want to allow a bot to save session data, and the other two scenarios will just outright cause problems if some one erringly returns the wrong value and those events do not occur. Support already has to deal with enough issues with hosts/servers blocking the session, so I don't want to give them another potential way to cause that problem.

Reply:
kk I'll go with that then